ci/build_wheel_libcugraph.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+# Copyright (c) 2025, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+package_name="libcugraph"
+package_dir="python/libcugraph"
+
+rapids-logger "Generating build requirements"
+matrix_selectors="cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION};cuda_suffixed=true"
+
+rapids-dependency-file-generator \
+  --output requirements \
+  --file-key "py_build_${package_name}" \
+  --file-key "py_rapids_build_${package_name}" \
+  --matrix "${matrix_selectors}" \
+| tee /tmp/requirements-build.txt
+
+rapids-logger "Installing build requirements"
+python -m pip install \
+    -v \
+    --prefer-binary \
+    -r /tmp/requirements-build.txt
+
+rapids-logger "Done build requirements"
+
+# build with '--no-build-isolation', for better sccache hit rate
+# 0 really means "add --no-build-isolation" (ref: https://github.com/pypa/pip/issues/5735)
+export PIP_NO_BUILD_ISOLATION=0
+
+RAPIDS_PY_CUDA_SUFFIX="$(rapids-wheel-ctk-name-gen ${RAPIDS_CUDA_VERSION})"
+
+PARALLEL_LEVEL=$(python -c \
+  "from math import ceil; from multiprocessing import cpu_count; print(ceil(cpu_count()/4))")
```

Comment:
This is copied from pylibcugraph, so I want to question it a bit. Are we really limited to only 1/4 of the available cores? I am guessing it's due to memory requirements at compile time?

Can we try to be more aggressive in a follow-up PR? Or can we add comments to explain why this is so limited?

Reply:
>  am guessing it's due to memory requirements at compile time?

Exactly, this was added back in https://github.com/rapidsai/cugraph/pull/4489#discussion_r1643392183 to deal with out-of-memory issues building wheels.

Since most of the heavy compilation would move to `libcugraph` wheels in this PR, I've moved this limiting (and requesting a `cpu32` node) over to `libcugraph` here.

HOWEVER... maybe now that these builds aren't also compiling RAFT and since https://github.com/rapidsai/cugraph/pull/4720 reduced the number of TUs being compiled, we could get away with a higher degree of parallelism.

I'll push a change here trying to remove this limit, let's see what happens.

Reply:
I just pushed a changed removing this limiting of `PARALLEL_LEVEL`. I left this requesting a `cpu32` node for now. Let's see how it goes.